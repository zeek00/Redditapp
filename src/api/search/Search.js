import {React , useState, Fragment} from 'react'
import styles from './Search.module.css'
import {FiSearch} from 'react-icons/fi'
import { useNavigate } from 'react-router-dom'


export default function Search() {
  const [focus, setFocus] = useState(false);
  const [searchPhrase, setSearchPhrase] = useState("")
  const [isMessage, setIsMessage] = useState(false)
  const navigate = useNavigate()
  const searchFocus = ()=>{
    console.log('focusing');
    setFocus(true)
    
  }


  const submitHandler = (event) => {
    event.preventDefault()
    if (searchPhrase.length >= 5){
      setIsMessage(false)
      console.log("search phrase is : " + JSON.stringify(searchPhrase))
      navigate(`/search/${searchPhrase}`)
      console.log("qq")
    }
    else {
      console.log("needed at least 5 characters, received: " + JSON.stringify(searchPhrase))
      setIsMessage(true)
    }

  }

  return (
    <Fragment>
      <form className={styles.container} onSubmit={submitHandler} aria-label="searchForm">
        <FiSearch className={focus ? styles.focus : styles.searchFont} />
        <input
            type="search"
            name="srch"
            id="srch"
            onFocus={searchFocus}
            onBlur={()=>setFocus(false)}
            onChange={(e)=>{setSearchPhrase(e.target.value)}}
            placeholder='Search-Posts' />
      </form>
      {isMessage && <p className={styles.message}>Please type in at least 5 characters</p>}
    </Fragment>

       
    
  )
}
