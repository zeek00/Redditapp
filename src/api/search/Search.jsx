import {React , useState} from 'react'
import styles from './Search.module.css'
import {FiSearch} from 'react-icons/fi'

export default function Search() {
  //const [searchPhrase, setSearchPhrase] = useState("")



//use form,when submit Navigate to /search/inputValue

  return (
    <div className={styles.container}>
        <FiSearch className={styles.searchFont} />
        <input type="search" name="" id="" placeholder='Search Posts' />
    </div>
  )
}
