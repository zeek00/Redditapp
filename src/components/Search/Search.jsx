import React from 'react'
import styles from './Search.module.css'
import {FiSearch} from 'react-icons/fi'

export default function Search() {
  return (
    <div className={styles.container}>
        <FiSearch className={styles.searchFont} />
        <input type="search" name="" id="" placeholder='Search Posts' />
    </div>
  )
}
