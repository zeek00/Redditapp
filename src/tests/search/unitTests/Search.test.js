import React from "react";
import {render, screen, fireEvent, act} from "@testing-library/react";
import Search from '../../../api/search/Search'
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";

const mockedNavigate = jest.fn().mockImplementation((calledWith) => {
        console.log("mockedNavigate was called " + JSON.stringify(calledWith))
    });

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedNavigate
}));
describe("Search", () => {
    beforeEach(() => {
        mockedNavigate.mockReset();
    });
    afterEach(() => {
        jest.restoreAllMocks();

    });

    it("user typed five characters, submitHandler and useNavigator were called with expected params", async () => {


        const searchResult = render(<Search/>)
        const submitHandlerMock = jest.fn()
        screen.getByRole("form", {name: "searchForm"}).onsubmit = submitHandlerMock

        await act(async () => {
            const input = screen.getByPlaceholderText('Search-Posts')
            userEvent.type(input, "mango{enter}");
        })

        const searchForm = screen.getByRole("form", {name: "searchForm"})
        expect(searchForm.nextElementSibling).toBe(null)
        expect(submitHandlerMock).toHaveBeenCalledTimes(1);
        expect(mockedNavigate).toHaveBeenCalledTimes(1);
        expect(mockedNavigate).toHaveBeenNthCalledWith(1, `/search/mango`);
        expect(searchResult).toMatchSnapshot()

    })

    it("user typed 0 characters, submitHandler called, useNavigator not called, message displayed", async () => {


        const searchResult = render(<Search/>)
        const submitHandlerMock = jest.fn()
        screen.getByRole("form", {name: "searchForm"}).onsubmit = submitHandlerMock

        await act(async () => {
            const input = screen.getByPlaceholderText('Search-Posts')
            userEvent.type(input, "{enter}");
        })
        const searchForm = screen.getByRole("form", {name: "searchForm"})
        console.dir(searchForm)


        expect(submitHandlerMock).toHaveBeenCalledTimes(1);
        expect(mockedNavigate).toHaveBeenCalledTimes(0);
        expect(searchResult).toMatchSnapshot()

    })

    it("user typed 1 characters, submitHandler called, useNavigator not called, message displayed", async () => {


        const searchResult = render(<Search/>)
        const submitHandlerMock = jest.fn()
        screen.getByRole("form", {name: "searchForm"}).onsubmit = submitHandlerMock

        await act(async () => {
            const input = screen.getByPlaceholderText('Search-Posts')
            userEvent.type(input, "m{enter}");
        })

        expect(submitHandlerMock).toHaveBeenCalledTimes(1);
        expect(mockedNavigate).toHaveBeenCalledTimes(0);
        expect(searchResult).toMatchSnapshot()

    })

   it("user clicked on input to make it onFocus, FiSearch className changes to 'focus'", async () => {


          const {container,  debug} = render(<Search/>)

       const searchForm = screen.getByRole("form", {name: "searchForm"})

          await act(async () => {
              const input = screen.getByPlaceholderText('Search-Posts')
              userEvent.click(input,undefined, {skipHover: true});

          })
        expect(searchForm.firstChild).toHaveClass("focus")
      })

      it("user , FiSearch className changes to 'focus'", async () => {


          const {container, debug} = render(<Search/>)
          const searchForm = screen.getByRole("form", {name: "searchForm"})

          await act(async () => {
              const input = screen.getByPlaceholderText('Search-Posts')
              userEvent.click(input,undefined, {skipHover: true});
          })
          await act(async () => {
              const inputForm = screen.getByRole("form", {name: "searchForm"})
              userEvent.click(inputForm, undefined, {skipHover: true})
          })

          expect(searchForm.firstChild).toHaveClass("searchFont")
    })



})
