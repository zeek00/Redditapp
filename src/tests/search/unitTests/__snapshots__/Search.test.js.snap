// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search user typed 0 characters, submitHandler called, useNavigator not called, message displayed 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        aria-label="searchForm"
        class="container"
      >
        <svg
          class="focus"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
        <input
          id="srch"
          name="srch"
          placeholder="Search-Posts"
          type="search"
        />
      </form>
      <p
        class="message"
      >
        Please type in at least 5 characters
      </p>
    </div>
  </body>,
  "container": <div>
    <form
      aria-label="searchForm"
      class="container"
    >
      <svg
        class="focus"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <line
          x1="21"
          x2="16.65"
          y1="21"
          y2="16.65"
        />
      </svg>
      <input
        id="srch"
        name="srch"
        placeholder="Search-Posts"
        type="search"
      />
    </form>
    <p
      class="message"
    >
      Please type in at least 5 characters
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Search user typed 1 characters, submitHandler called, useNavigator not called, message displayed 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        aria-label="searchForm"
        class="container"
      >
        <svg
          class="focus"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
        <input
          id="srch"
          name="srch"
          placeholder="Search-Posts"
          type="search"
        />
      </form>
      <p
        class="message"
      >
        Please type in at least 5 characters
      </p>
    </div>
  </body>,
  "container": <div>
    <form
      aria-label="searchForm"
      class="container"
    >
      <svg
        class="focus"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <line
          x1="21"
          x2="16.65"
          y1="21"
          y2="16.65"
        />
      </svg>
      <input
        id="srch"
        name="srch"
        placeholder="Search-Posts"
        type="search"
      />
    </form>
    <p
      class="message"
    >
      Please type in at least 5 characters
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Search user typed five characters, submitHandler and useNavigator were called with expected params 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        aria-label="searchForm"
        class="container"
      >
        <svg
          class="focus"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
        <input
          id="srch"
          name="srch"
          placeholder="Search-Posts"
          type="search"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      aria-label="searchForm"
      class="container"
    >
      <svg
        class="focus"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <line
          x1="21"
          x2="16.65"
          y1="21"
          y2="16.65"
        />
      </svg>
      <input
        id="srch"
        name="srch"
        placeholder="Search-Posts"
        type="search"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
